tenseal/tensealcontext.cpp

```diff
@@ -2,30 +2,27 @@
 
 #include <seal/seal.h>
 
-#include <vector>
-
 #include "utils.h"
 
+namespace tenseal {
+
 using namespace seal;
 using namespace std;
 
-namespace tenseal {
-
 TenSEALContext::TenSEALContext(EncryptionParameters parms) {
     this->_parms = parms;
     this->_context = SEALContext::Create(parms);
 
     KeyGenerator keygen = KeyGenerator(this->_context);
 
-    this->_public_key =
-        shared_ptr<PublicKey>(new PublicKey(keygen.public_key()));
-    this->_secret_key =
-        shared_ptr<SecretKey>(new SecretKey(keygen.secret_key()));
-    this->encryptor = shared_ptr<Encryptor>(
-        new Encryptor(this->_context, *this->_public_key));
-    this->decryptor = shared_ptr<Decryptor>(
-        new Decryptor(this->_context, *this->_secret_key));
-    this->evaluator = shared_ptr<Evaluator>(new Evaluator(this->_context));
+    this->_public_key = make_shared<PublicKey>(keygen.public_key());
+    this->_secret_key = make_shared<SecretKey>(keygen.secret_key());
+    this->encryptor =
+        make_shared<Encryptor>(this->_context, *this->_public_key);
+    this->decryptor =
+        make_shared<Decryptor>(this->_context, *this->_secret_key);
+    this->evaluator = make_shared<Evaluator>(this->_context);
+    this->encoder = make_shared<TenSEALEncoder>(this->_context);
```

Comment:
I just learned about the advantages of using make_shared, and it looks even better